### PR TITLE
Add unit tests for org.apache.rocketmq.common.UtilAll

### PR DESCRIPTION
**Pull request by EricHetti**
I've analysed your codebase and noticed that `org.apache.rocketmq.common.UtilAll` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: common/pom.xml [hunk 1]
```diff
@@ -37,5 +37,16 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>rocketmq-remoting</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito</artifactId>
+            <version>1.6.6</version>
```
Comment (vongosling) [inline common/pom.xml]: Have you noticed the dependency mockito in the parent pom? powermock is a good extension for mockito. I strongly recommend that you could move the dependency into parent pom and scope it in the test.

File: common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+public class UtilAllTests {
+
+    @PrepareForTest({RuntimeMXBean.class,
+            UtilAll.class, ManagementFactory.class})
+    @Test
+    public void testGetPid1() {
```
Comment (vongosling) [inline common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java]: testMethodwithCondtion is a recommended naming not simple 1, 2,3...

File: common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        Assert.assertEquals(-1,
+                UtilAll.getDiskPartitionSpaceUsedPercent("foo"), 0);
+    }
+
+    @Rule
+    private TemporaryFolder folder = new TemporaryFolder();
+
+    @Test
+    public void testGetDiskPartitionSpaceUsedPercent1() throws IOException {
+        File file = folder.newFile("foo.txt");
```
Comment (vongosling) [inline common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java]: Do you consider the data clearance when test stateful function?

**Comment by vongosling** [comment]
Anyway, make the class utils to be testable is a valuable thing. I am very glad to help you to polish your pr and merge to branch finally if you could follow up my comments:-)

File: common/pom.xml (hunk 1 above)
Comment (Braavos96) [reply common/pom.xml]: PowerMock has been moved to the parent pom and version has been changed

File: common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java (hunk 2 above)
Comment (Braavos96) [reply common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java]: Test method names have been changed

File: common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java (hunk 3 above)
Comment (Braavos96) [reply common/src/test/java/org/apache/rocketmq/common/UtilAllTests.java]: @vongosling I have added a finally block that deletes the file and TemporaryFolder. Is that ok? I am not entirely sure about that

**Comment by Braavos96** [comment]
Hi @vongosling just following up and wondering if you would like me to make any changes to the PR?

File: pom.xml [hunk 4]
```diff
@@ -471,6 +471,17 @@
             <version>2.6.3</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (liuruiyiyang) [inline pom.xml]: Can we implement these tests using existing test-utils dependencies?   It is required to check the license of the newly added dependencies and keep our dependency tree clean and each item is necessary.

Comment (Braavos96) [reply pom.xml]: Hi, Yes I can do it with `mockito`, however `powermock` is a bit more powerful and you will be losing coverage if I used `mockito` instead

**Pull request merged**